### PR TITLE
Resume downloading from previous state.

### PR DESCRIPTION
**Pull request by mit456**
1. Continue downloading from the point it stopped. If it got stopped due
to some interruptions.

### CODE REVIEW

**Comment by mit456** [comment]
@Bartzi  Modified download_fsns.py to resume downloading, See if it can be useful.

File: datasets/fsns/download_fsns.py [hunk 1]
```diff
@@ -34,19 +34,27 @@
...
-
-					downloaded += len(buffer)
-					f.write(buffer)
-					print("Got: {:>10} of {:>10} bytes".format(downloaded, file_size), end='\r')
-
-			print("{}".format(" " * 100), end='\r')
\ No newline at end of file
+			file_size = int(urllib.request.urlopen(url).info()['Content-Length'])
+
+			if not os.path.exists(os.path.join(dest_dir, file_name)) or\
```
Comment (Bartzi) [inline datasets/fsns/download_fsns.py]: could you change this to:
```python
if (not os.path.exists(os.path.join(dest_dir, file_name)) or 
    os.stat(os.path.join(dest_dir, file_name)).st_size != file_size):
```

Comment (mit456) [reply datasets/fsns/download_fsns.py]: Done. Thanks for the suggestion.

File: datasets/fsns/download_fsns.py [hunk 2]
```diff
@@ -34,19 +34,27 @@
...
-
-					downloaded += len(buffer)
-					f.write(buffer)
-					print("Got: {:>10} of {:>10} bytes".format(downloaded, file_size), end='\r')
-
-			print("{}".format(" " * 100), end='\r')
\ No newline at end of file
+			file_size = int(urllib.request.urlopen(url).info()['Content-Length'])
+
+			if (not os.path.exists(os.path.join(dest_dir, file_name)) or\
```
[the remark this answers is not in the document]
Comment (mit456) [reply datasets/fsns/download_fsns.py]: My bad, Done. But it works with \ too. Correct me if I am wrong

**Comment by mit456** [comment]
One question I would like to ask, I should not ask here but still asking, Is it okay to test trained fsns model on an unknown image, does that make sense or should I train on my dataset? because I tried testing on a random image and it throws me an error.

**Comment by Bartzi** [comment]
nice :smile:

**Pull request merged**